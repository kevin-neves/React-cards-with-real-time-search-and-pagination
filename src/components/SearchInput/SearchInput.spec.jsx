import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SearchInput } from '.';

describe('<SearchInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<SearchInput handleChange={fn} searchValue={'testando'} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<SearchInput handleChange={fn} searchValue="alguma coisa" />);

    const input = screen.getByPlaceholderText(/type your search/i);
    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe('alguma coisa');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    const { container } = render(<SearchInput handleChange={fn} searchValue="" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
