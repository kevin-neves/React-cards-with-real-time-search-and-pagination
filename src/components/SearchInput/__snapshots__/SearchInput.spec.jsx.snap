// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should have a value of searchValue 1`] = `
<input
  class="text-input"
  placeholder="Type your search"
  type="search"
  value=""
/>
`;
